### PR TITLE
Kaushan inactive page

### PR DESCRIPTION
**Pull request by Kaushan05**
inactive employee details table finished

### CODE REVIEW

File: src/app/employee-details/inactive-employee/inactive-employee.component.html [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      <ng-template pTemplate="body" let-lefted>
+        <tr>
+          <td><img id="inactive_employe_img" src=" lefted.Photo" /></td>
+          <td>{{ lefted.employeeName }}</td>
+          <td>{{ lefted.EmployeeNo }}</td>
+          <td>{{ lefted.deportment }}</td>
+          <td>
+            <p-button
+              styleClass="p-button-warning p-button-sm"
+              (onClick)="indiviActiveFn(lefted)"
```
Comment (HarishDass) [inline src/app/employee-details/inactive-employee/inactive-employee.component.html:43]: Give some meaning full names

File: src/app/employee-details/inactive-employee/inactive-employee.component.html [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+              >Inactive</p-button
+            >
+          </td>
+          <td>
+            <p-button
+              label="Edit"
+              styleClass="p-button-sm"
+              (click)="activateFn(lefted)"
```
Comment (HarishDass) [inline src/app/employee-details/inactive-employee/inactive-employee.component.html:51]: Give some meaning full names

File: src/app/employee-details/inactive-employee/inactive-employee.component.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  clearthebar: string = '';
+
+  constructor(private LinkData: LeftoverTableService) {}
+  ngOnInit() {
+    this.LinkData.employeeDataFn().subscribe((res: EmployeData[]) => {
+      this.leftoverdatas = res.filter((resp) => resp.Active == false);
+    });
+  }
+
+  activateFn(lefted: EmployeData) {
```
Comment (HarishDass) [inline src/app/employee-details/inactive-employee/inactive-employee.component.ts:26]: this function has no functionality

File: src/app/employee-details/inactive-employee/inactive-employee.component.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    this.individualData = lefted;
+    this.EmpnameFordis = lefted.employeeName;
+    this.activating = '';
+  }
+
+  toggleActiveFn() {
+    if (this.activating == 'true') {
+      this.individualData.Active = true;
+      this.LinkData.updateDataFn(this.individualData).subscribe(
+        (x: EmployeData) => {
```
Comment (HarishDass) [inline src/app/employee-details/inactive-employee/inactive-employee.component.ts]: Give some meaning full names

File: src/app/employee-details/inactive-employee/inactive-employee.component.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+          const updatedDetails = this.leftoverdatas.filter(
+            (x1: EmployeData) => x.id !== x1.id
```
Comment (HarishDass) [inline src/app/employee-details/inactive-employee/inactive-employee.component.ts]: Give some meaning full names

File: src/app/employee-details/inactive-employee/inactive-employee.component.ts [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  templateUrl: './inactive-employee.component.html',
+  styleUrls: ['./inactive-employee.component.css'],
+})
+export class InactiveEmployeeComponent {
+  leftoverdatas: EmployeData[] = [];
+  visible: boolean = false;
+  individualData: any = {};
+  disabled = false;
+  activating!: string;
+  EmpnameFordis: string = '';
```
Comment (HarishDass) [inline src/app/employee-details/inactive-employee/inactive-employee.component.ts]: remove this variable there isn't any need for this variable

File: src/app/employee-details/inactive-employee/inactive-employee.component.ts [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+  activateFn(lefted: EmployeData) {
+    console.log(lefted);
+    this.visible = false;
+    this.activating = '';
+  }
+
+  indiviActiveFn(lefted: EmployeData) {
+    this.visible = true;
+    this.individualData = lefted;
+    this.EmpnameFordis = lefted.employeeName;
```
Comment (HarishDass) [inline src/app/employee-details/inactive-employee/inactive-employee.component.ts]: remove this variable there isn't any need for this variable

File: src/app/employee-details/inactive-employee/inactive-employee.component.html [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+      <ng-template pTemplate="emptymessage">
+        <tr>
+          <td colspan="6" id="no_details_text">
+            No Employee found with this detail.
+          </td>
+        </tr>
+      </ng-template>
+    </p-table>
+    <p-dialog
+      header="Do you want to Activate :- {{ EmpnameFordis }}"
```
Comment (HarishDass) [inline src/app/employee-details/inactive-employee/inactive-employee.component.html:65]: EmpnameFordis is not needed try to solve it in different way

File: src/app/employee-details/inactive-employee/inactive-employee.component.html [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+            styleClass="p-button-sm"
+            (onClick)="toggleActiveFn()"
+          ></p-button>
+        </div>
+      </div>
+    </p-dialog>
+  </p-card>
+</div>
+
+<!--
```
Comment (HarishDass) [inline src/app/employee-details/inactive-employee/inactive-employee.component.html]: remove this commented lines of code

File: src/app/employee-details/inactive-employee/inactive-employee.component.ts [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+@Component({
+  selector: 'app-inactive-employee',
+  templateUrl: './inactive-employee.component.html',
+  styleUrls: ['./inactive-employee.component.css'],
+})
+export class InactiveEmployeeComponent {
+  leftoverdatas: EmployeData[] = [];
+  visible: boolean = false;
+  individualData: any = {};
+  disabled = false;
```
Comment (HarishDass) [inline src/app/employee-details/inactive-employee/inactive-employee.component.ts]: declared but not used

File: src/app/employee-details/inactive-employee/inactive-employee.component.ts [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+  activating!: string;
+  EmpnameFordis: string = '';
+  clearthebar: string = '';
```
Comment (HarishDass) [inline src/app/employee-details/inactive-employee/inactive-employee.component.ts]: declared but not used

File: src/app/employee-details/inactive-employee/inactive-employee.component.ts [hunk 12]
```diff
@@ -0,0 +1,55 @@
+import { Component } from '@angular/core';
+import { EmployeData } from 'src/app/shared-module/employe-data';
+import { LeftoverTableService } from 'src/app/Services/leftover-table.service';
+import { filter } from 'rxjs';
```
Comment (HarishDass) [inline src/app/employee-details/inactive-employee/inactive-employee.component.ts]: declared but never used

File: src/app/employee-details/inactive-employee/inactive-employee.component.ts [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+export class InactiveEmployeeComponent {
+  leftoverdatas: EmployeData[] = [];
+  visible: boolean = false;
+  individualData: any = {};
+  activating!: string;
+
+  constructor(private LinkData: LeftoverTableService) {}
+  ngOnInit() {
+    this.LinkData.employeeDataFn().subscribe((res: EmployeData[]) => {
+      this.leftoverdatas = res.filter((resp) => resp.Active == false);
```
Comment (HarishDass) [inline src/app/employee-details/inactive-employee/inactive-employee.component.ts]: give some meaningful name resp

**Comment by Kaushan05** [comment]
Screenshot 2023-10-04 104839
Screenshot 2023-10-04 104938

**Review by HarishDass — APPROVED** [review]

**Review by aarthi1246 — APPROVED** [review]

**Review by HarishDass — APPROVED** [review]

File: src/app/home/home.module.ts [hunk 14]
```diff
@@ -1,10 +1,17 @@
 import { NgModule } from '@angular/core';
 import { CommonModule } from '@angular/common';
+
```
Comment (bharathidhasan2002) [inline src/app/home/home.module.ts]: Remove white Space

File: package-lock.json [hunk 15]
```diff
@@ -250,12 +250,12 @@
       }
     },
     "node_modules/@angular-devkit/schematics": {
```
Comment (bharathidhasan2002) [inline package-lock.json]: don't push the package-lock.json

File: src/app/employee-details/inactive-employee/inactive-employee.component.ts [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+  leftoverdatas: EmployeData[] = [];
+  visible: boolean = false;
+  individualData: any = {};
+
+  constructor(private LinkData: LeftoverTableService) {}
+  ngOnInit() {
+    this.GetDataFn();
+  }
+
+  GetDataFn() {
```
Comment (bharathidhasan2002) [inline src/app/employee-details/inactive-employee/inactive-employee.component.ts]: change function name

File: src/app/employee-details/inactive-employee/inactive-employee.component.ts [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+    this.LinkData.employeeDataFn().subscribe((Firstresponce: EmployeData[]) => {
```
Comment (bharathidhasan2002) [inline src/app/employee-details/inactive-employee/inactive-employee.component.ts]: change employeeDataFn name

File: src/app/employee-details/inactive-employee/inactive-employee.component.ts [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+      this.leftoverdatas = Firstresponce.filter(
```
Comment (bharathidhasan2002) [inline src/app/employee-details/inactive-employee/inactive-employee.component.ts]: change the variable name

File: src/app/employee-details/inactive-employee/inactive-employee.component.ts [hunk 19]
```diff
@@ -0,0 +1,50 @@
...
+        (responce) => responce.Active == false
+      );
+    });
+  }
+
+  indiviActiveFn(lefted: EmployeData) {
```
Comment (bharathidhasan2002) [inline src/app/employee-details/inactive-employee/inactive-employee.component.ts]: Remove Fn

Comment (bharathidhasan2002) [inline src/app/employee-details/inactive-employee/inactive-employee.component.ts]: change the name lefted name

File: src/app/employee-details/inactive-employee/inactive-employee.component.ts [hunk 20]
```diff
@@ -0,0 +1,50 @@
...
+import { LeftoverTableService } from 'src/app/shared-module/Services/leftover-table.service';
+@Component({
+  selector: 'app-inactive-employee',
+  templateUrl: './inactive-employee.component.html',
+  styleUrls: ['./inactive-employee.component.css'],
+})
+export class InactiveEmployeeComponent {
+  leftoverdatas: EmployeData[] = [];
+  visible: boolean = false;
+  individualData: any = {};
```
Comment (bharathidhasan2002) [inline src/app/employee-details/inactive-employee/inactive-employee.component.ts:12]: Remove any

File: src/app/employee-details/inactive-employee/inactive-employee.component.ts [hunk 21]
```diff
@@ -0,0 +1,50 @@
...
+    });
+  }
+
+  indiviActiveFn(lefted: EmployeData) {
+    this.visible = true;
+    this.individualData = lefted;
+    console.log(this.individualData);
+  }
+
+  toggleActiveFn() {
```
Comment (bharathidhasan2002) [inline src/app/employee-details/inactive-employee/inactive-employee.component.ts]: Change the Function name

**Review by HarishDass — APPROVED** [review]

**Pull request merged**